kore/src/Kore/Strategies/Verification.hs

```diff
@@ -233,26 +237,32 @@ verifyClaim
   =
     traceExceptT D_OnePath_verifyClaim [debugArg "rule" goal] $ do
     let
-        startPattern = ProofState.Goal $ getConfiguration goal
+        startPattern =
+            ProofState.Goal (ProofState.ExecutionDepth 0)
+                $ getConfiguration goal
         limitedStrategy =
             strategy goal claims axioms
             & Foldable.toList
             & Limit.takeWithin depthLimit
-    handle
-        handleLimitExceeded
-        $ Strategy.leavesM
-            updateQueue
-            (Strategy.unfoldTransition transit)
-            (limitedStrategy, startPattern)
-            & fmap discardStrategy
-            & throwUnproven
+        proofStatesLogicT =
+            Strategy.leavesM
+                updateQueue
+                (Strategy.unfoldTransition transit)
+                (limitedStrategy, startPattern)
+                & fmap discardStrategy
+        result = handle handleLimitExceeded $ proofStatesLogicT & throwUnproven
+
+    proofStateList <- result
+    infoProofLength proofStateList
+
+    Monad.void result
```

Comment:
Won't this compute the results twice?

Where is the depth reported when the proof fails?